plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -966,6 +969,11 @@ private static ReaderPageSourceWithRowPositions createParquetPageSource(
             dataSource = createDataSource(inputFile, OptionalLong.of(fileSize), options, memoryContext, fileFormatDataSourceStats);
             ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
             FileMetaData fileMetaData = parquetMetadata.getFileMetaData();
+            // For Hive and Parquet case, when Hive writes to Parquet, it normalizes timestamp to UTC and the timezone writes to the footer
+            DateTimeZone parquetDateTimeZone = Optional.ofNullable(fileMetaData.getKeyValueMetaData().get("writer.time.zone"))
```

Comment:
Do we have migrate tests with Hive that have `writer.time.zone` in the Parquet footer, as well as for Hive tabes that do not?

(any data written before hive 3.1.something probably does not have `writer.time.zone` in the footer, right)

Reply:
Let's bring some facts together:

Hive:
Historically, timestamp in Parquet had Instant semantics.
The community wanted to have consistency across different file format, so Hive 3.1 turned off normalisation to UTC.
Hive 3.1.2  restored the practice of normalising to UTC and also introduced recording the session-local local time zone in the file metadata (writer.time.zone in the footer)

In this way:
Newer version of Hive converts timestamp from UTC to the saved time zone (instead of local time zone).
Older version of Hive (not aware of writer.time.zone)  read timestamp in UTC (historically Instant semantics) and converts to local time zone.

Hive does the conversion only if file's metadata has createdby field's value starting with "parquet-mr".


Spark:
When Iceberg do the migration: 
Spark read the timestamp value and converts it to local session time zone.

Trino cannot read INT96 timestamp after  Iceberg.migrate function was used as the IN96 Timestamp is mapped to Timestamp(6) not supported by Trino.


Trino:
Trino has own Parquet reader and writer which is used by both cases when Trino deals with Hive and Iceberg catalogs.
Trino uses "parquet-mr-trino version…" as createdby value in file metadata.
Trino adds "writer.time.zone" value when writing timestamp in Hive catalog ( It does not do this when writing to Iceberg)


In tests:
Trino uses local timezone when dealing with Hive.

Reply:
> For the case when Hive timestamp without writer.time.zone in the footer, how should we approach?

fail migration, or have `hive.parquet.time-zone` setting.
ie the values read by iceberg later on, should be the same (nor "somewhat similar") to the value read by Hive connector prior to the migration

Reply:
I added  a change that now we fail migration when there is timestamp without timezone column type and there is no writer.time.zone in the footer.  PTAL

Reply:
@findepi can you please take a look?

Reply:
I don't agree with the approach to try to mimic Apache Hive behaviour in iceberg connector.
We aren't looking at the timezone in parquet footer during reads even in our hive connector, so it's inconsistent to start doing it in iceberg now.
I think hive tables migrated to iceberg should follow iceberg spec and not be expected to exactly match hive behaviour (e.g. we drop hive bucketing when migrating bucketed hive tables to iceberg).
I would prefer that we convert these columns into TIMESTAMP WITH TIMEZONE type. That will result in these columns getting read explicitly as timestamp with UTC timezone with no scope for ambiguity. This is the behaviour that we saw from running Apache Spark's migrate procedure as well in https://github.com/trinodb/trino/issues/11338.
This will force users to deal with the underlying problem by either opting to alter table schemas to ignore timezone or updating queries, rather than continuing to rely on Hive timestamp behaviour under the covers.